fdbserver/workloads/Mako.actor.cpp

```diff
@@ -0,0 +1,534 @@
+		operationsSpec = getOption(options, LiteralStringRef("operations"), LiteralStringRef("g100")).contents().toString();
+		//  parse the sequence and extract operations to be executed
+		parseOperationsSpec();
+		for (int i = 0; i < MAX_OP; ++i) {
+			// initilize per-operation latency record
+			opLatencies.push_back(opLatencies.arena(), ContinuousSample<double>(rowCount / sampleSize));
+			// initialize per-operation counter
+			opCounters.push_back(PerfIntCounter(opNames[i]));
+		}
+	}
+
+	std::string description() override {
+		// Mako is a simple workload to measure the performance of FDB.
+		// The primary purpose of this benchmark is to generate consistent performance results
+		return "Mako";
+	}
+
+	Future<Void> setup(Database const& cx) override {
+		// use all the clients to populate data
+		if (populateData)
+			return _setup(cx, this);
+		return Void();
+	}
+
+	Future<Void> start(Database const& cx) override {
+		return _start(cx, this);
+	}
+
+	Future<bool> check(Database const& cx) override {
+		return true;
+	}
+
+	void getMetrics(vector<PerfMetric>& m) override {
+		// metrics of population process
+		if (populateData){
+			m.push_back( PerfMetric( "Mean load time (seconds)", loadTime, true ) );
+			// The importing rate of keys, controlled by parameter "insertionCountsToMeasure"
+			auto ratesItr = ratesAtKeyCounts.begin();
+			for(; ratesItr != ratesAtKeyCounts.end(); ratesItr++){
+				m.push_back(PerfMetric(format("%ld keys imported bytes/sec", ratesItr->first), ratesItr->second, false));
+			}
+		}
+		// benchmark
+		if (runBenchmark){
+			m.push_back(PerfMetric("Measured Duration", testDuration, true));
+			m.push_back(xacts.getMetric());
+			m.push_back(PerfMetric("Transactions/sec", xacts.getValue() / testDuration, true));
+			m.push_back(totalOps.getMetric());
+			m.push_back(PerfMetric("Operations/sec", totalOps.getValue() / testDuration, true));
+			m.push_back(conflicts.getMetric());
+			m.push_back(PerfMetric("Conflicts/sec", conflicts.getValue() / testDuration, true));
+			m.push_back(retries.getMetric());
+
+			// count of each operation
+			for (int i = 0; i < MAX_OP; ++i){
+				m.push_back(opCounters[i].getMetric());
+			}
+
+			// Meaningful Latency metrics
+			const int opExecutedAtOnce[] = {OP_GETREADVERSION, OP_GET, OP_GETRANGE, OP_SGET, OP_SGETRANGE, OP_COMMIT};
+			for (const int& op : opExecutedAtOnce){
+				m.push_back(PerfMetric("Mean " + std::string(opNames[op]) +" Latency (ms)", 1000 * opLatencies[op].mean(), true));
+				m.push_back(PerfMetric("Max " + std::string(opNames[op]) + " Latency (ms, averaged)", 1000 * opLatencies[op].max(), true));
+				m.push_back(PerfMetric("Min " + std::string(opNames[op]) + " Latency (ms, averaged)", 1000 * opLatencies[op].min(), true));
+			}
+
+			//insert logging metrics if exists
+			m.insert(m.end(), periodicMetrics.begin(), periodicMetrics.end());
+		}
+	}
+	static void randStr(std::string& str, const int& len) {
```

Comment:
If you're going to mutate the std::string, please take a pointer to it rather than a mutable reference.  (It'd be more C++-y to just return an std::string instead, but I'm assuming you're trying to keep this function looking like the one below.)

Reply:
Done.